src/NServiceBus.Metrics/RawData/LongValueWriter.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace NServiceBus.Metrics.RawData
+{
+    using System;
+    using System.IO;
+
+    static class LongValueWriter
+    {
+        const long Version = 1;
+
+        // WIRE FORMAT, Version: 1
+
+        // 0                   1                   2                   3
+        // 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+        //+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        //|     Version   | Min date time | date1 |    value 1    | date2 |  
+        //+---------------+---------------+-------------------------------+
+        //|     value 2   | date3 |   value  3    | date4 |     value 4   |
+        //+---------------+---------------+-------------------------------+
+
+        public static void Write(BinaryWriter outputWriter, ArraySegment<RingBuffer.Entry> chunk)
+        {
+            var array = chunk.Array;
+            var offset = chunk.Offset;
+            var count = chunk.Count;
+
+            Array.Sort(array, offset, count, EntryTickComparer.Instance);
+
+            var minDate = array[offset].Ticks;
+
+            outputWriter.Write(Version);
+            outputWriter.Write(minDate);
+
+            for (var i = 0; i < count; i++)
+            {
+                // int allows to write ticks of 7minutes, as reporter runs much more frequent, this can be int
+                var date = (int)(array[offset + i].Ticks - minDate);
```

Comment:
I don't understand `Again,` as I saw no comment related to this before. Let me try to answer a question that I assume you asked: "why is this a local var for the array".

Indexer `this[int index]` in an array segment is provided only for `IList<T>`. Having this said, any operation on a segment would require to load the segment, then access its property. I haven't measured this, but this looked like a faster version to me. If it isn't or it's clumsy, I can inline all the vars.